questionpy_server/settings.py

```diff
@@ -44,20 +51,20 @@ def __call__(self, settings: BaseSettings) -> dict[str, Any]:
 class GeneralSettings(BaseModel):
     log_level: Literal['NONE', 'DEBUG', 'INFO', 'WARNING', 'ERROR', 'CRITICAL'] = 'INFO'
 
-    @validator('log_level', pre=True)
-    # pylint: disable=no-self-argument
+    @field_validator('log_level', mode="before")
+    @classmethod
     def logging_level_to_upper(cls, value: str) -> str:
         return value.upper()
 
 
 class WebserviceSettings(BaseModel):
     listen_address: str = '127.0.0.1'
     listen_port: int = 9020
-    max_main_size: ByteSize = Field(ByteSize(5 * MiB), const=True)
+    max_main_size: ByteSize = ByteSize(5 * MiB)
```

Comment:
`max_main_size`  darf nur genau 5MiB sein.
Man könnte das über `typing.Literal` realisieren: `max_main_size: Literal[ByteSize(12)] = ByteSize(5 * MiB)`, allerdings ist das im Vergleich zu `Field(..., const=True)` keine so schöne Lösung.

Reply:
Warum ist das denn Teil der Settings, wenn es nicht anpassbar sein soll?

Reply:
Es war als fixe Einstellung gedacht, um konsistent mit `max_package_size` und damit schneller auffindbar zu sein. Man könnte es aber auch woanders platzieren.

Reply:
Hm, ja ich würde das eher als Konstante in `web.py` sehen. Aber ich mache mal einen `Literal` draus.

Reply:
Bzw. `ClassVar`, damit ich mich nicht mit den Warnungen, dass in `Literal` kein Objekt reingehört, herumschlagen muss.